src/server/v2.0/handler/replication.go

```diff
@@ -113,6 +113,14 @@ func (r *replicationAPI) CreateReplicationPolicy(ctx context.Context, params ope
 		policy.CopyByChunk = *params.Policy.CopyByChunk
 	}
 
+	if params.Policy.SingleActiveReplication != nil {
+		// Validate and assign SingleActiveReplication only for non-event_based triggers
+		if params.Policy.Trigger.Type == model.TriggerTypeEventBased && *params.Policy.SingleActiveReplication == true {
```

Comment:
Is this necessary?  If we don't allow single active replication for event-based replication, we should also validate it in 
`UpdateReplicationPolicy`

Also a nit: `*params.Policy.SingleActiveReplication == true`  -> `*params.Policy.SingleActiveReplication`